src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -388,6 +400,32 @@ void GcsPlacementGroupManager::HandleGetAllPlacementGroup(
   ++counts_[CountType::GET_ALL_PLACEMENT_GROUP_REQUEST];
 }
 
+void GcsPlacementGroupManager::HandleWaitPlacementGroupUntilReady(
+    const rpc::WaitPlacementGroupUntilReadyRequest &request,
+    rpc::WaitPlacementGroupUntilReadyReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  PlacementGroupID placement_group_id =
+      PlacementGroupID::FromBinary(request.placement_group_id());
+  RAY_LOG(DEBUG) << "Waiting for placement group until ready, placement group id = "
+                 << placement_group_id;
+
+  // If the placement group is not exist, return directly.
+  if (!registered_placement_groups_.contains(placement_group_id)) {
+    GCS_RPC_SEND_REPLY(send_reply_callback, reply,
+                       Status::Invalid("Placement group is not exist."));
+  }
+
```

Comment:
maybe we should check whether the state is already created from `registered_placement_groups_ `? if not, the callback maybe never invoked.

Reply:
Yeah, you are right, thanks.